connectors/src/connectors/zendesk/lib/zendesk_api.ts

```diff
@@ -240,115 +240,99 @@ export async function fetchRecentlyUpdatedArticles({
     url:
-      `https://${brandSubdomain}.zendesk.com/api/v2/help_center/categories/${categoryId}/articles?` +
-      (cursor ? `page[after]=${encodeURIComponent(cursor)}&` : "") +
-      `page[size]=${pageSize}`,
+      url ?? // using the URL if we got one, reconstructing it otherwise
+      `https://${brandSubdomain}.zendesk.com/api/v2/help_center/categories/${category.categoryId}/articles?page[size]=${pageSize}`,
     accessToken,
   });
   return (
-    response || { articles: [], meta: { has_more: false, after_cursor: "" } }
+    response || { articles: [], meta: { has_more: false }, links: { next: "" } }
   );
 }
 
 /**
  * Fetches a batch of the recently updated tickets from the Zendesk API using the incremental API endpoint.
  */
-export async function fetchRecentlyUpdatedTickets({
-  brandSubdomain,
-  accessToken,
-  startTime = null,
-  cursor = null,
-}: // pass either a cursor or a start time, but not both
-| {
-      brandSubdomain: string;
-      accessToken: string;
-      startTime: number | null;
-      cursor?: never;
-    }
-  | {
-      brandSubdomain: string;
-      accessToken: string;
-      startTime?: never;
-      cursor: string | null;
-    }): Promise<{
+export async function fetchRecentlyUpdatedTickets(
+  accessToken: string,
+  {
+    brandSubdomain,
+    startTime,
+    url,
+  }:
+    | { brandSubdomain: string; startTime: number; url?: never }
+    | { brandSubdomain?: never; startTime?: never; url: string }
+): Promise<{
   tickets: ZendeskFetchedTicket[];
   end_of_stream: boolean;
-  after_cursor: string;
+  after_url: string | null;
 }> {
   const response = await fetchFromZendeskWithRetries({
     url:
-      `https://${brandSubdomain}.zendesk.com/api/v2/incremental/tickets/cursor.json` +
-      (cursor ? `?cursor=${encodeURIComponent(cursor)}` : "") +
-      (startTime ? `?start_time=${startTime}` : ""),
+      url ?? // using the URL if we got one, reconstructing it otherwise
+      `https://${brandSubdomain}.zendesk.com/api/v2/incremental/tickets/cursor.json?start_time=${startTime}`,
     accessToken,
   });
-  return (
-    response || {
-      tickets: [],
-      end_of_stream: false,
-      after_cursor: "",
-    }
-  );
+  return response || { tickets: [], end_of_stream: false, after_url: "" };
 }
 
 /**
  * Fetches a batch of tickets from the Zendesk API.
  * Only fetches tickets that have been solved, and that were updated within the retention period.
  */
-export async function fetchZendeskTicketsInBrand({
-  brandSubdomain,
-  accessToken,
-  pageSize,
-  cursor,
-  retentionPeriodDays,
-}: {
-  brandSubdomain: string;
-  accessToken: string;
-  pageSize: number;
-  cursor: string | null;
-  retentionPeriodDays: number;
-}): Promise<{
-  tickets: ZendeskFetchedTicket[];
-  meta: { has_more: boolean; after_cursor: string };
+export async function fetchZendeskTicketsInBrand(
+  accessToken: string,
+  {
+    brandSubdomain,
+    pageSize,
+    retentionPeriodDays,
+    url,
+  }:
+    | {
```

Comment:
is this not the same as:
```
{ brandSubdomain: string;  pageSize: number; retentionPeriodDays: number;} | {url: string}
````
?

Reply:
in both cases, some of the properties won't exist at all no? sorry if this is super dumb haven't been working in ts for very long 🙏

Reply:
Merging to see if this fixes the issue with 422 errors, but let's talk about this again, I may be missing smth